### PR TITLE
use parse for BigFloat/BigInt

### PR DESCRIPTION
**Pull request by simonbyrne**
This deprecates `BigInt/BigFloat(::AbstractString)` methods, in favour of `parse` (see #10594 and #10599 ). 
- It also defines `big_str` and `bigfloat_str` macros (analogous to the `bigint_str` macro): on that note, would these be better as `BigFloat_str` and `BigInt_str`?
- I changed the default "base" argument in `parse(BigFloat,)` to `0`: this is basically the same as `10`, but allows for use of `0x`/`0b` prefixes for hex-float/bit-float (#9371) formats.

### CODE REVIEW

**Comment by dpsanders** [comment]
Nice. Could we also extend the big function so we can write

big("0.1")

? 

I am constantly writing this...

File: base/deprecated.jl [hunk 1]
```diff
@@ -529,3 +529,6 @@ export float32_isvalid, float64_isvalid
 @deprecate parseint(T::Type, s, base) parse(T, s, base)
 
 @deprecate linrange linspace
+
+@deprecate BigFloat(s::AbstractString) parse(BigFloat,AbstractString)
+@deprecate BigInt(s::AbstractString) parse(BigInt,AbstractString)
```
Comment (tkelman) [inline base/deprecated.jl]: should these be
```jl
@deprecate BigFloat(s::AbstractString) parse(BigFloat,s)
@deprecate BigInt(s::AbstractString) parse(BigInt,s)
```
?

Comment (simonbyrne) [inline base/deprecated.jl]: Good catch, thanks!

**Comment by simonbyrne** [comment]
I've made it a string macro, so you can write
```
big"0.1"
```

Is that sufficient, or would a function be useful as well?

**Comment by simonbyrne** [comment]
Okay, I've replaced `bigint_str` with `bit_str`.

**Comment by StefanKarpinski** [comment]
LGTM. @JeffBezanson?

**Comment by mschauer** [comment]
Small inconsistency: ``tryparse()`` allows a base argument, but ``parse`` not.

**Comment by simonbyrne** [comment]
Ah, good point: I was just exposing the GMP/MPFR functionality. Our `parse` & `tryparse`functions for other types don't support different bases either.

**Comment by mschauer** [comment]
For some Int they do,
``
julia> parse(Int,"1A",16)
26
``

**Comment by simonbyrne** [comment]
Ah, good point. We should expose this then.

**Pull request merged**